src/main/java/seedu/algobase/model/Id.java

```diff
@@ -0,0 +1,49 @@
+package seedu.algobase.model;
+
+import java.util.UUID;
+
+/**
+ * Unique ID for AlgoBase models.
+ */
+public class Id {
+    private final String idValue;
+
+    private Id(String idValue) {
+        this.idValue = idValue;
+    }
+
+    /**
+     * Generate Id based on unix timestamp.
+     */
+    public static Id generateId() {
+        return new Id(UUID.randomUUID().toString());
```

Comment:
Does UUID enforce uniqueness over time? Update the header by the way.

Reply:
Good enough https://stackoverflow.com/questions/20999792/does-randomuuid-give-a-unique-id